src/frontend/src/optimizer/plan_node/logical_changelog.rs

```diff
@@ -66,15 +66,26 @@ impl PlanTreeNodeUnary for LogicalChangeLog {
         input_col_change: ColIndexMapping,
     ) -> (Self, ColIndexMapping) {
         let changelog = Self::new(input, self.core.need_op, true);
-        if self.core.need_op {
+
+        let out_col_change = if self.core.need_op {
             let mut output_vec = input_col_change.to_parts().0.to_vec();
             let len = input_col_change.to_parts().1;
             output_vec.push(Some(len));
-            let out_col_change = ColIndexMapping::new(output_vec, len + 1);
-            (changelog, out_col_change)
+            ColIndexMapping::new(output_vec, len + 1)
+        } else {
+            input_col_change
+        };
+
+        let mut output_vec = out_col_change.to_parts().0.to_vec();
+        let len = out_col_change.to_parts().1;
+        let out_col_change = if self.base.stream_key().is_none() {
+            ColIndexMapping::new(output_vec, len + 1)
         } else {
-            (changelog, input_col_change)
-        }
+            output_vec.push(Some(len));
+            ColIndexMapping::new(output_vec, len + 1)
+        };
```

Comment:
I don't get this logic, probably because I am not familiar with the optimzier codes. But it seems that we are relying on some implicit assumption here. Can you elaborate more?
- Why do we need to extend the `ColIndexMapping` without extending `output_vec` when the stream key is none? 
- When will the stream key be none?

Reply:
stream_key.is_none means that _changelog_row_id is not included in the select result. This has been changed to need_changelog_row_id  to make the logic clearer.
When the results does not contain _changelog_row_id , _changelog_row_id is the hidden column added by this operator, and the ColIndexMapping should be null->index (here is len)